### PR TITLE
Expose failed summaries with idempotent two-stage retry

### PR DESCRIPTION
**Pull request by ellaaicare**
## Summary

Backend slices for ellaaicare/ella-ai#1050 and the historical recovery contract in ellaaicare/ella-ai#1051.

- expose preserved `conversation_summary_failed` records through authenticated conversation history
- add authenticated, metadata-only `GET /v1/conversations/{conversation_id}/processing-retry-plan`
- add authenticated `POST /v1/conversations/{conversation_id}/processing-retries` with `{request_id, correction_text?}`
- atomically claim exact UID-owned records with lease/attempt fencing and replay-safe durable receipts
- fetch/decrypt the transcript only after the claim, receipt its SHA-256 without returning transcript content, and fail closed if it changes
- recover a missing generic summary through the configured stock OMI provider, then persist and vector it independently
- run stage 2 directly through the shared Hermes API-session path with the stable canonical per-user session and full lossless transcript
- strictly confirm the enriched OMI version, canonical ledger write, and enriched vector metadata before terminal success
- preserve a usable generic summary and retry only the failed enrichment/canonical/vector phase

## Exact historical behavior

A completed generic conversation with missing Hermes enrichment is classified as `enrichment_only`. It does not rerun stock summarization. If canonical writeback already succeeded, replay reuses the same trace/version and retries only canonical confirmation or the enriched vector.

Pinecone vector existence is not treated as freshness. Generic and enriched vectors carry `active_summary_version_id` and `summary_content_sha256`; the enriched phase force-upserts and verifies both values. Receipt transactions also fence the active summary version and retry attempt.

## Safety

- UID comes only from Firebase auth; clients cannot select another profile or Firestore path.
- Metadata planning performs zero writes and returns no transcript text.
- Active processing, intentional discard, locked, non-retryable, and cross-user/not-found states fail closed.
- A 15-minute lease permits same-request reclaim after worker loss while rejecting concurrent attempts.
- No n8n/OpenClaw/provision-shim dependency is added to this retry path.
- No notifications, integrations, extraction, usage accounting, batch repair, production write, or deployment is included.
- Raw provider errors are not returned to clients.

## Validation

- `./backend/test.sh`: 116 passed
- `test_ella_conversation_corrections.py`: 46 passed
- `test_conversation_processing_retry.py`: 19 passed
- `test_conversation_processing_failures.py`: 9 passed
- `test_ella_callbacks_summary_update.py`: 13 passed
- `test_llm_openrouter_provider.py`: 3 passed
- `python3 -m py_compile` for all changed backend modules: passed
- `git diff --check`: clean

## Deployment gate

Merge and deploy the OMI backend before the iOS retry UI is promoted. Do not run the #1051 historical production repair without a separate exact-record dry-run and explicit approval. The known `f541...` fixture remains stage-2-only; this PR has not mutated it or any production record.

### CODE REVIEW

**Comment by ellaaicare** [comment]
AgentStatus: needs-work

Backend review found release blockers. This path touches protected transcripts and durable summary state, and the current implementation cannot repair the exact known record.

1. Completed + missing Hermes enrichment is treated as terminal at backend/database/conversations.py:920-930. The known record f5416025-acc5-4def-8c25-7d8c837ebedf is already completed with a generic summary/vector but has no enriched summary provenance. It needs a stage-2-only claim that preserves the generic version and retries Hermes/writeback/vector/canonical provenance.

2. Generic completion is persisted before Hermes at backend/ella/services/summary_recovery.py:359-371. A crash or Hermes failure leaves the conversation completed, while a later retry is rejected as already complete. Persist explicit generic_status and enrichment_status; completed+enrichment pending/failed must remain retryable without rerunning generic/vector.

3. FastAPI BackgroundTasks is not a durable queue. A restart after the Firestore claim leaves outcome=processing indefinitely, and replaying the same request_id just returns processing without re-enqueue. Add a lease/attempt timestamp and replay-safe reclaim/requeue after expiry, or route work through an existing durable worker. Include a crash-between-claim-and-worker test.

4. The transaction returns the raw encrypted/compressed Firestore document, then the route constructs Conversation(**conversation_data) after claiming. This can raise and strand the claim. Return only claim metadata and fetch/decrypt with get_conversation after commit, or decode before model construction. This is also flagged at discussion_r3619323207.

5. The recovery prompt permits category=media, but CategoryEnum does not. Use the actual enum values or normalize media to entertainment. See discussion_r3619323210.

6. The response contract needs separate generic and enrichment states plus replay/lease semantics. A generic success with Hermes pending must be visible as such, not indistinguishable from fully completed.

Required tests: protected transcript claim, completed generic + missing enrichment, Hermes failure then successful stage-2 replay, stale lease reclaim, same request replay under transport uncertainty, different request while leased, and no duplicate summary versions/vectors/canonical events. Keep unrelated notifications/integrations/n8n/usage disabled.

Do not run generic recovery again for f541; it is the stage-2-only approval fixture.

**Comment by ellaaicare** [comment]
Additional review invariant for the revision:

An enriched OMI version plus canonical confirmation is not terminal until the enriched conversation vector is also confirmed. If `save_structured_vector()` fails after the enriched version/canonical write, the current draft leaves `enrichment_state=writeback_applied`; the next claim can classify the conversation as `already_enriched` and never repair the missing/stale vector.

Please persist a separate enrichment/vector phase in the receipt. Replay must reuse the same enriched version/canonical trace and retry only the vector, then mark fully completed. The metadata-only plan should include a SHA-256 of the authoritative preserved transcript/source (not only UID/session hashes). Keep the existing requested tests for stale lease reclaim/requeue and post-claim decrypted fetch; those remain release gates.

**Comment by ellaaicare** [comment]
Exact-fixture blocker in the current local design: vector **existence** is not sufficient for the enriched-vector phase. `f541...` already has the generic-summary vector, so an `_ensure_conversation_vector()` helper that returns when the conversation ID exists would skip re-embedding after Hermes changes the active summary and then falsely mark the enriched vector complete.

For the enriched phase, either persist/compare an active-summary-version or content hash in vector metadata, or unconditionally call the idempotent vector upsert once per new enriched version and verify the stored metadata matches that version/hash. The receipt should record the enriched version/hash it indexed. Generic-stage presence checks may reuse a matching generic vector, but cannot stand in for enriched-vector freshness.

Also apply attempt fencing to every receipt phase transaction, not only source recording, so a delayed prior attempt using the same request UUID cannot advance the reclaimed attempt's phase.

**Comment by ellaaicare** [comment]
Additional release blocker from backend review:

`save_structured_vector()` is not a summary-only vector operation. It calls `retrieve_metadata_fields_from_transcript()`, which invokes `llm_mini` over the full transcript before upserting Pinecone. Historical recovery therefore currently re-runs unrelated entity/topic/date extraction and hidden provider usage, violating the approved no-unrelated-side-effects/usage boundary. On the exact 25,775-character fixture this is especially undesirable.

For recovery, add/use a purpose-built summary-vector refresh that:
- embeds only the current `structured` summary;
- preserves the existing conversation vector metadata;
- overwrites only the vector values plus `active_summary_version_id` and `summary_content_sha256` (and required uid/memory_id keys);
- does not call transcript metadata extraction, notifications/timezone lookup, integrations, or usage accounting;
- force-upserts once for a new enriched version, then confirms the exact version/hash;
- skips on replay only when that exact version/hash is already confirmed.

Please add a test that fails if `retrieve_metadata_fields_from_transcript` is called during generic/enriched historical recovery, and verify the vector metadata preservation contract. This remains a merge blocker alongside the freshness receipt.

**Comment by ellaaicare** [comment]
AgentStatus: ready-for-review

Requested-changes blockers are addressed in commit `bbe900bd0`.

- Completed generic + missing enrichment now claims `enrichment_only`; stock generation is not rerun.
- Receipt/API state is split into generic writeback, generic vector, enrichment/canonical, and enriched vector phases.
- Claims use a 15-minute lease plus incrementing attempt count. An expired same-ID receipt is reclaimed/requeued; a live same/different request is not duplicated.
- Claim transactions return metadata only. The worker fetches/decrypts the UID-owned conversation after claim and receipts only transcript SHA-256/count.
- Hermes recovery uses the direct API session with `canonical_omi_session_key(uid)` and the complete transcript; it has no n8n, provision-shim, or OpenClaw dependency.
- Category output is normalized against `CategoryEnum`.
- Strict writeback does not report success until the canonical writer returns `ok=true`. Pending canonical replay reuses the same trace/version.
- Enriched vector success now requires an idempotent upsert followed by verification of both `active_summary_version_id` and structured-content SHA-256. A pre-existing generic vector cannot satisfy it.
- Every receipt phase uses attempt fencing. Final vector and terminal receipt phases also fence the active summary version.
- Tests cover stale lease reclaim, active lease conflicts, protected transcript loading, transport uncertainty, stage-2-only replay, no model rerun after canonical success, generic/enriched vector freshness, vector failure replay, canonical idempotency, and concurrent summary changes.

Validation after formatting:
- `./backend/test.sh`: 116 passed
- corrections: 46 passed
- retry transactions: 19 passed
- processing failures: 9 passed
- callback writeback: 13 passed
- OpenRouter provider: 3 passed
- py_compile and diff check: passed

No production record was read through the new endpoint or mutated. The `f541...` exact repair remains separately gated after review, merge, deploy, authenticated zero-write plan, and explicit approval.

**Comment by ellaaicare** [comment]
AgentStatus: needs-work

The ready-for-review update does not address the summary-vector side-effect blocker posted immediately before it: recovery still calls `save_structured_vector()`, which invokes `retrieve_metadata_fields_from_transcript()` / `llm_mini` over the full transcript.

Before merge, replace that recovery call with a summary-only vector refresh that preserves existing metadata, embeds only `structured`, updates/validates the active summary version and content hash, and never invokes transcript metadata extraction, notification/timezone lookup, integrations, or usage accounting. Add an executable negative test proving the transcript metadata extractor is not called.

Keep the branch open and post the follow-up commit/tests for re-review.

**Comment by ellaaicare** [comment]
AgentStatus: needs-work

Exact production metadata-only verification found a fixture-specific blocker before any write:

- status=completed, discarded=false
- generic structured summary present
- 99 preserved segments
- summary_version_count=0
- active_summary_version_id=null
- no enrichment/canonical state

The current `_existing_generic_version_id()` returns the sentinel `existing-generic-summary` for this valid legacy state, but the later active-version guard compares that sentinel to the real null active ID and will fail with `conversation_summary_changed`. That means #288 still cannot repair the approved exact fixture.

Handle legacy generic summaries without regenerating them:
- preserve null active ID as a valid baseline;
- fence a no-version legacy baseline by structured-summary content hash (plus the existing transcript hash), not by a fake version ID;
- if stock generic recovery actually creates a version, fence by that real version ID/hash;
- let strict Hermes writeback bootstrap the legacy generic version and append the enriched version atomically through the existing `build_summary_version_update()` path;
- add an exact-shape test: completed + usable structured + zero versions + null active ID => `enrichment_only`, no stock provider call, no generic overwrite, successful legacy bootstrap + enriched active version.

Do not expose the UID or transcript. Production remains zero-write so far.

**Comment by ellaaicare** [comment]
AgentStatus: ready-for-review

The no-unrelated-vector-side-effects blocker is addressed in `e7314e786`.

Recovery no longer calls `save_structured_vector()`. The new `refresh_structured_summary_vector()`:
- serializes and embeds only the current structured summary;
- reads and preserves all existing Pinecone metadata;
- overwrites only vector values plus authoritative `uid`, `memory_id`, `active_summary_version_id`, and `summary_content_sha256`;
- performs no transcript metadata/entity/topic/date extraction, timezone/notification lookup, integrations, or usage accounting;
- uses the existing conversation timestamp only when no vector `created_at` metadata exists.

The `enrichment_only` path reuses an existing generic vector without rewriting it. After canonical Hermes writeback, the enriched phase still force-upserts exactly once and confirms the enriched version/hash before terminal success.

New regression: `test_conversation_summary_vector.py` injects failures into transcript metadata extraction and timezone lookup, verifies neither is called, verifies embedding input is summary-only JSON, and verifies metadata preservation plus protected key replacement.

Validation:
- summary-vector regression: 1 passed
- corrections/recovery: 46 passed
- retry transactions: 19 passed
- processing failures: 9 passed
- callback writeback: 13 passed
- OpenRouter provider: 3 passed
- backend smoke: 116 passed
- py_compile/diff check: passed

No production mutation or deployment was performed.

**Comment by ellaaicare** [comment]
AgentStatus: ready-for-review

The exact unversioned-legacy fixture blocker is addressed in `c69a44c20`.

For a completed conversation with a usable structured summary, zero `summary_versions`, and null `active_summary_version_id`:

- recovery remains `enrichment_only`;
- the internal legacy discriminator is no longer compared to the null active version as if it were a real ID;
- the authoritative structured-summary SHA-256 is recorded in the durable retry receipt alongside the transcript SHA-256;
- the worker fences the null-version baseline by both hashes and requires the active version to remain null before Hermes generation;
- the direct Hermes service rechecks transcript hash, structured-summary hash, retry attempt, and active-version baseline immediately before strict apply;
- stock generic generation is not called and the existing generic vector is not rewritten;
- `build_summary_version_update()` bootstraps the preserved generic summary as inactive `legacy_current`, then appends the new `recovered_enriched` active version based on that real bootstrap ID.

The metadata-only plan now also reports `structured_summary_sha256` without returning transcript or summary content.

New executable coverage proves:
- the exact fixture shape reaches Hermes without generic regeneration/vector overwrite;
- both source-hash receipts remain stable;
- legacy version history contains the preserved generic base plus enriched active version.

Validation:
- recovery/API: 47 passed
- receipt/transaction/version bootstrap: 20 passed
- summary-vector side-effect safety: 1 passed
- failure/list API: 9 passed
- callback writeback: 13 passed
- provider: 3 passed
- backend smoke: 116 passed
- py_compile/diff check: passed

Production remains zero-write and undeployed.

**Comment by ellaaicare** [comment]
Backend manager review complete on c69a44c20.

No remaining blocker found. The exact legacy fixture is now supported as `enrichment_only` without regenerating its stock summary; transcript/timestamps are fenced and preserved; Hermes uses the canonical per-user API session; canonical and enriched-vector phases are independently confirmed; and retries are idempotent with attempt/lease fencing.

Independent validation:
- focused recovery/correction/vector/callback/provider suites: 93 passed
- backend smoke: 116 passed
- diff check: clean
- branch: clean and mergeable

Risk remains high because this touches durable summary/version state, so production rollout remains a one-record canary: authenticated zero-write plan first, exact `f541...` repair second, before/after metadata and side-effect receipts, then no wider batch without review.

AgentStatus: approved

**Comment by ellaaicare** [comment]
AgentStatus: done

Router reconciliation for stale `pull_request/opened` callback:

- PR #288 completed backend review and merged as `28e7a404f0975f50375697a7033f27172cf77103`.
- Follow-ups #289 and #290 merged the live side-effect-preservation and terminal-receipt fixes.
- The merged backend was deployed and health-verified.
- Exact canary `f5416025-acc5-4def-8c25-7d8c837ebedf` completed through `enrichment_only` recovery with transcript/timestamps preserved, canonical Hermes confirmation, enriched vector confirmation, and no broad `/reprocess` side effects.
- Final exhaustive fleet inventory found 0 additional qualifying long failed/stale/empty-summary candidates.
- iOS retry UI is merged and shipped in TestFlight `1.0.525 (794)`.
- Parent issues ella-ai#1049 and ella-ai#1050 are closed completed.

The Kanban chain remains complete. No merge, re-dispatch, or reopening is needed.

File: backend/database/conversations.py [hunk 1]
```diff
@@ -890,6 +893,380 @@ def update_conversation_status(uid: str, conversation_id: str, status: str):
...
+    retry_ref,
+    request_id: str,
+    requested_at: datetime,
+):
+    snapshot = conversation_ref.get(transaction=transaction)
+    retry_snapshot = retry_ref.get(transaction=transaction)
+    if not snapshot.exists:
+        return {'outcome': 'not_found', 'conversation': None}
+
+    conversation = snapshot.to_dict()
```
[the remark this answers is not in the document]
Comment (ellaaicare) [reply backend/database/conversations.py:991]: Resolved before merge. The claim transaction now returns receipt/claim metadata only; after commit, the worker performs a UID-owned `get_conversation()` fetch through the normal decode/decrypt path before constructing the `Conversation` model. The lease/attempt receipt remains reclaimable if post-claim loading fails, so this path cannot leave an unqueued permanent busy claim. Executable protected-transcript and stale-lease tests passed in the final 93-test focused suite. Final implementation head `c69a44c20`; merged as `28e7a404f0975f50375697a7033f27172cf77103`, deployed and canary-validated. This callback is stale; no reopening or re-dispatch is needed.

File: backend/ella/services/summary_recovery.py [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+    return f"""You are Ella, the user's companion summary writer.
+
+Recover an OMI conversation whose first summary attempt failed. Use the complete transcript and durable companion context available in this Hermes session. Return JSON only.
+
+Rules:
+- Produce one accurate, warm, specific summary grounded in the transcript.
+- overview must start with "[Ella] ".
+- Do not invent details or expose raw speaker labels when a natural description is available.
+- title must be short and contain no markdown.
+- category should be one of: personal, family, education, health, technology, work, business, finance, legal, media, music, news, travel, other.
```
[the remark this answers is not in the document]
Comment (ellaaicare) [reply backend/ella/services/summary_recovery.py]: Resolved before merge. Recovery output is now normalized against the actual `CategoryEnum`; unsupported `media` is not passed into strict writeback and media-like output is mapped to the supported category contract. The final recovery/API suite covered category normalization and passed as part of the 93 focused tests. Final head `c69a44c20`; merged as `28e7a404f0975f50375697a7033f27172cf77103`, deployed and canary-validated.

**Comment by ellaaicare** [comment]
AgentStatus: done

Router reconciliation for stale aggregate review `4741092187`:

- The automated review targeted obsolete commit `8ff5c20642`.
- Both findings were addressed on final head `c69a44c20`: protected transcripts are fetched/decrypted only after the metadata-only claim, and recovery categories are normalized against the actual `CategoryEnum`.
- Both inline threads now have fix receipts and are resolved.
- PR #288 merged as `28e7a404f0975f50375697a7033f27172cf77103`, was deployed, and passed the exact production canary.

No Kanban change, merge, re-dispatch, or reopening is required.
